protobuf/sync/v1/sync_service.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package sync.v1;
+
+option go_package = "flagd/grpcsync";
+
+/*
+   SyncFlagsRequest convey a key that identifies the flagd instance, enabling Sync provider to filter
+   specific flags for a specific flagd deployment.
+*/
+message SyncFlagsRequest {
+  string key = 1;
```

Comment:
Should we do better than `key`? We want to make sure not to create confusion between this and `flagKey`. What prefix would best describe what this is the key for... `configurationKey`?

Reply:
Is this a flag key? If so, why not just use `flagKey`?

Reply:
> Is this a flag key? If so, why not just use `flagKey`?

It's precisely not a flag key.

Reply:
Yeah, I guess a more descriptive name and maybe a comment would help here.

Reply:
Yes, agree with the requirement to change the naming here. 

Meaning should focus on "provider identifier" so, I think `providerId` fits better

Reply:
I changed the name and added more description 🤝